### ISSUE TITLE
_CURRENT_BFF_DIR_ is empty string

### ISSUE THREAD

**Issue by billw2012** [fastbuild/fastbuild#666]
I think it should be the full path to the directory even if its the cwd, or it needs validation before using it directly in places that won't accept an empty string.

This gives an error for instance:

![image](https://user-images.githubusercontent.com/1453936/69902005-c4664600-1380-11ea-8ef1-d0bb52d059e8.png)

![image](https://user-images.githubusercontent.com/1453936/69902002-bc0e0b00-1380-11ea-8e87-409b8f5350bf.png)

![image](https://user-images.githubusercontent.com/1453936/69902014-d34cf880-1380-11ea-92f2-f7ee7b795db2.png)

**Comment by belkiss** [comment]
Agreed! But instead of full path it should be replaced by a single '.'

In the meantime, we worked around it by prefixing all occurrences of `$_CURRENT_BFF_DIR_$` by `.\`, like this:
```
.UnityOutputPath = '.\$_CURRENT_BFF_DIR_$\..\tmp\blob\mathlib'

**Comment by john-blackburn** [comment]
I noticed this too and had to use `_WORKING_DIR_` instead. That one works giving the full path to the current working directory. I always run fbuild in the fbuild.bff directory so that's OK for me. But for those who run from another directory `_CURRENT_BFF_DIR_` needs to be fixed.

**Comment by ffulin** [comment]
Having it use "." (without a slash) instead of empty for the root directory makes the most sense to me. This is consistent (in terms of not having a trailing slash) and should be backwards compatible for existing uses I can think of.

**Comment by to01z** [comment]
A related issue to this discussion is how one might go about obtaining a full path to the current dir (which may be needed when invoking some custom exes). `_CURRENT_BFF_DIR_` is expressed as a relative path to `_WORKING_DIR_` (always a rooted path) which means the two can be concated to form a rooted path. This, however, falls apart if the current dir is on a different drive than the bff file on Windows (in which case both vars are rooted and can't be concatenated).

I wonder if it might make sense to introduce a `_CURRENT_BFF_DIR_FULL_` variable which always contains a fully rooted path for such situations.
